pkg/skaffold/deploy/kubectl.go

```diff
@@ -339,7 +344,12 @@ func recursiveReplaceImage(i interface{}, replacements map[string]*replacement)
 	}
 }
 
-func removeTag(image string) string {
-	re := regexp.MustCompile(":[^/]+$")
-	return re.ReplaceAllString(image, "")
+func splitTag(image string) (string, string) {
+	re := regexp.MustCompile(`(.*):([^/]+)$`)
+	matches := re.FindStringSubmatch(image)
```

Comment:
It might be a little more robust to reuse the name parsing utilities here: https://github.com/google/go-containerregistry/blob/master/name/ref.go